### DISCUSSION TITLE
Support for Maven repo in Google Artifact Registry

### DISCUSSION THREAD

**Discussion by ohlen1** [renovatebot/renovate#15251] [Request Help] [answered]
Hi,

We run a self hosted renovatebot in GKE. The Maven repo resides in the same GCP project as the GKE cluster.
Has anyone successfully got renovate to scan for new packages within a Maven repo hosted in Google Artifact Registry? I have not really been able to figure out what additional configuration is required, if any. If a POM depends on modules in this Maven repo it contains information about repo URL etc. Is Renovate using this information to figure out how to access the repos, or is additional information required in renovatebot configuration?

Thanks!

**Comment by ohlen1** [comment, marked as answer]
I'll answer my own question, in case anyone is scratching their head because of the same topic.
It worked for us by using username/password authentication through a GCP service account.
1. Create SA with role `roles/artifactregistry.reader`
2. In Renovate config, add `hostRule`
```
{
   "hostType": "maven",
   "matchHost": "europe-west4-maven.pkg.dev",
   "username": "_json_key_base64",
   "password": <base64 encoded SA key>
}
```
and `packageRule`:
```
{"matchManagers": ["maven"], "registryUrls": ["https://<gcp region>-maven.pkg.dev/<gcp project name>/<repo name>"] }
```

**Comment by trietsch** [comment]
@ohlen1 this approach has been working fine for me until recently. Now Renovate is complaining about:
```
DEBUG: Unsupported Maven protocol (repository=myrepository)
       "url": "artifactregistry://europe-west4-maven.pkg.dev/my-gcp-project/release/com/zaxxer/HikariCP/maven-metadata.xml"
```

Any ideas? This is due to the fact that the notation in my Gradle build files uses `artifactregistry://` to configure the Maven artifact repositories, but I find it strange that this has worked before.

**Reply by trietsch** [reply]
Got it working again, as I'm using Gradle, this needed to be added to the `matchManagers`. In the past, `maven` alone was enough. `packageRule`:
```
{
  matchManagers: [
    "maven",
    "gradle"
  ],
  registryUrls: [
    "https://repo1.maven.org/maven2",
    "https://europe-west4-maven.pkg.dev/my-gcp-project/release"
  ]
}
```

and `hostRule`:
```
{
  matchHost: "https://europe-west4-maven.pkg.dev/my-gcp-project/release/",
  username: "_json_key_base64",
  password: "{{ secrets.ARTIFACT_REGISTRY_MAVEN_RELEASE_PASSWORD }}"
}
```

With the secret notation, you can configure Renovate secrets to prevent having secrets in your config file. Configure the environment variable `RENOVATE_SECRETS` with the following value, i.e. do (needs to be a valid JSON so you could configure more secrets if needed):
```
export RENOVATE_SECRETS="{\"ARTIFACT_REGISTRY_MAVEN_RELEASE_PASSWORD\":\"base64 encoded password\"}"
```

**Reply by trietsch** [reply]
Ensure that the host rule does not contain a `hostType` by the way, if you're also using Gradle.
